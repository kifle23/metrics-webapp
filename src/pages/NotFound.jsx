import React from 'react';

const NotFound = () => (
  <div>Page not found!</div>
);

export default NotFound;
