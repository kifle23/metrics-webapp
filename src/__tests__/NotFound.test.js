import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import NotFound from '../pages/NotFound';

describe('NotFound page', () => {
  test('renders without errors', () => {
    const component = renderer.create(
      <Router>
        <NotFound />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the "Page not found!" element', () => {
    render(
      <Router>
        <NotFound />
      </Router>
    );
    const notFoundElement = screen.getByText('Page not found!');
    expect(notFoundElement).toBeInTheDocument();
  });
});
