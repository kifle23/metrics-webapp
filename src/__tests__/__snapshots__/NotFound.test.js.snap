// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create header Render without errors 1`] = `
<div>
  Page not found!
</div>
`;

exports[`NotFound page renders without errors 1`] = `
<div>
  Page not found!
</div>
`;
